airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/http_response_filter.py

```diff
@@ -22,25 +25,43 @@ class HttpResponseFilter(JsonSchemaMixin):
         http_codes (Set[int]): http code of matching requests
         error_message_contains (str): error substring of matching requests
         predicate (str): predicate to apply to determine if a request is matching
+        error_message (Union[InterpolatedString, str): error message to display if the response matches the filter
     """
 
     TOO_MANY_REQUESTS_ERRORS = {429}
     DEFAULT_RETRIABLE_ERRORS = set([x for x in range(500, 600)]).union(TOO_MANY_REQUESTS_ERRORS)
 
     action: Union[ResponseAction, str]
+    config: Config
     options: InitVar[Mapping[str, Any]]
     http_codes: Set[int] = None
     error_message_contains: str = None
     predicate: Union[InterpolatedBoolean, str] = ""
+    error_message: Union[InterpolatedString, str] = ""
 
     def __post_init__(self, options: Mapping[str, Any]):
         if isinstance(self.action, str):
             self.action = ResponseAction[self.action]
         self.http_codes = self.http_codes or set()
         if isinstance(self.predicate, str):
             self.predicate = InterpolatedBoolean(condition=self.predicate, options=options)
+        self.error_message = InterpolatedString.create(string_or_interpolated=self.error_message, options=options)
 
-    def matches(self, response: requests.Response) -> Optional[ResponseAction]:
+    def matches(self, response: requests.Response, backoff_time: Optional[float] = None) -> Optional[ResponseStatus]:
```

Comment:
Refactored `matches()` to directly construct `ResponseStatus` instead of `ResponseAction`. This was mainly to keep message creation internal to the filter instead of passing things in between the error handler and tighten the interface